### PR TITLE
READY: "Reset Bucket Properties" command

### LINKED ISSUE #142: Add reset bucket properties command. [JIRA: CLIENTS-793]
[`riak_pb`](https://github.com/basho/riak_pb/blob/develop/src/riak.proto#L73-L77)

### PR DESCRIPTION
**Pull request by lukebakken**
Resolves #142 (CLIENTS-783) (CLIENTS-793)

### CODE REVIEW

File: lib/core/riakconnection.js [hunk 1]
```diff
@@ -94,10 +94,12 @@ function RiakConnection(options) {
     }
 
     this._emitAndClose = function(evt, evt_args) {
-        this.closed = true;
-        this._connection.end();
-        this.emit(evt, this, evt_args);
-        this.close();
+        if (!this.closed) {
```
Comment (alexmoore) [inline lib/core/riakconnection.js]: Is this is for the case where the connection dies ?

File: test/integration/kv/fetchandsetbuckettypeprops.js [hunk 2]
```diff
@@ -46,7 +40,7 @@ describe('Store and Fetch Bucket props - Integration', function() {
             cluster.execute(store);
         });
     
-        it('Should store bucket props for a non-default bucket type', function(done) {
+        it('should-store-for-non-default', function(done) {
```
Comment (alexmoore) [inline test/integration/kv/fetchandsetbuckettypeprops.js]: What prompted all the test name renaming?

File: test/integration/kv/fetchpreflist.js [hunk 3]
```diff
@@ -14,24 +12,26 @@ var cb = function (err, resp) {
 
 describe('FetchPreflist - Integration', function() {
     var cluster;
-    this.timeout(10000);
-
     before(function(done) {
-        var nodes = RiakNode.buildNodes(Test.nodeAddresses);
-        cluster = new RiakCluster({ nodes: nodes});
-        cluster.start(function (err, rslt) {
+        cluster = Test.buildCluster(function (err, rslt) {
```
Comment (alexmoore) [inline test/integration/kv/fetchpreflist.js]: Love this code reduction.

File: lib/core/riakconnection.js (hunk 1 above)
Comment (lukebakken) [inline lib/core/riakconnection.js]: `_emitAndClose` just raises whatever event is passed to it and closes the connection. Several other events trigger this method, including cases where a connection attempt times out (raises `connectFailed`) or a connection is unexpectedly closed (`connectionClosed`)

File: test/integration/kv/fetchandsetbuckettypeprops.js (hunk 2 above)
Comment (lukebakken) [inline test/integration/kv/fetchandsetbuckettypeprops.js]: `mocha` (the test runner) accepts an `--fgrep` argument to only run tests matching the grep. Using dashes makes it more shell friendly.

**Comment by alexmoore** [comment]
Code looks good, tests pass.  +1

**Pull request merged**